android/app/src/main/java/org/fossasia/openevent/core/track/session/SessionViewHolder.java

```diff
@@ -1,66 +1,282 @@
+
+        setSessionStatus();
+
+        Track track = session.getTrack();
+
+        if (!RealmDataRepository.isNull(track)) {
+            int storedColor = Color.parseColor(track.getColor());
+
+            if (type != trackWiseSessionList) {
+                color = storedColor;
+            } else {
+                color = colorInTracks;
+            }
+
+            TextDrawable drawable = Views.getTextDrawableBuilder().round()
+                    .build(String.valueOf(track.getName().charAt(0)), storedColor);
+
+            trackImageIcon.setImageDrawable(drawable);
+            trackImageIcon.setBackgroundColor(Color.TRANSPARENT);
+            sessionTrack.setText(track.getName());
+
+        } else {
+            trackImageIcon.setVisibility(View.GONE);
+            sessionTrack.setVisibility(View.GONE);
+            Timber.d("This session has a null or incomplete track somehow : " + session.getTitle() + " " + track);
+        }
+
+        itemView.setOnClickListener(v -> handleItemClicks.itemOnClick(getAdapterPosition(), getLayoutPosition()));
+
+        String date = DateConverter.formatDateWithDefault(DateConverter.FORMAT_DATE_COMPLETE, session.getStartsAt());
+        sessionDate.setText(date);
+        sessionTime.setText(String.format("%s - %s",
+                DateConverter.formatDateWithDefault(DateConverter.FORMAT_12H, session.getStartsAt()),
+                DateConverter.formatDateWithDefault(DateConverter.FORMAT_12H, session.getEndsAt())));
+
+        if (session.getMicrolocation() != null) {
+            String locationName = Utils.checkStringEmpty(session.getMicrolocation().getName());
+            sessionLocation.setText(locationName);
+        } else {
+            sessionLocation.setText(context.getString(R.string.location_not_decided));
+        }
+
+        setObservable();
+
+        handleVisibilityByType(type);
+
+        setBookmarkClickListener(realmRepo, track);
+
+        // Set color generated by palette on views
+        sessionHeader.setBackgroundColor(color);
+        if (track != null && track.isValid()) {
+            sessionTitle.setTextColor(Color.parseColor(track.getFontColor()));
+            setBookmarkIcon(sessionBookmarkIcon, session.getIsBookmarked(), track.getFontColor());
+        }
+    }
+
+    private void setObservable() {
+        Observable.just(session.getSpeakers())
+                .map(speakers -> {
+                    ArrayList<String> speakerName = new ArrayList<>();
+
+                    for (Speaker speaker : speakers) {
+                        String name = Utils.checkStringEmpty(speaker.getName());
+                        speakerName.add(name);
+                    }
+
+                    if (speakers.isEmpty()) {
+                        sessionSpeaker.setVisibility(View.GONE);
+                        speakerIcon.setVisibility(View.GONE);
+                    }
+
+                    return TextUtils.join(", ", speakerName);
```

Comment:
Actually you are not using Observable here at all. All things are happening in serial. It's the same as writing it all imperatively with a for loop for Speakers, If you do want to use functional approach you may and it is better, but write it  completely

```Java
Observable.just(session.getSpeakers())
                .map(speakers ->
```

is just like 

```Java
List<Speaker> speakers = session.getSpeakers();
List<Speaker> yetAnotherSpeakers = speakers;
```

Write it like this:

```Java
Observable.fromIterable(session.getSpeakers())
    .map(speaker.getName())
    .map(Utils::checkStringEmpty) // Also, what is checkStringEmpty rename it to mean what it does
    .toList()
    .map(names -> TextUtils.join(", ", names))
    .subscribe(sessionSpeaker::setText);
```

And you are checking `if (speakers.isEmpty())` after creating n observable and iterating through speakers. Doesn't make sense. Add the check before and only do stuff if it is not empty